Source/Core/Common/DynamicLibrary.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/DynamicLibrary.h"
+#include <cstring>
+#include "Common/Assert.h"
+#include "Common/StringUtil.h"
+
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <dlfcn.h>
+#endif
+
+namespace Common
+{
+DynamicLibrary::DynamicLibrary() = default;
+
+DynamicLibrary::DynamicLibrary(const char* filename)
+{
+  Open(filename);
+}
+
+DynamicLibrary::~DynamicLibrary()
+{
+  Close();
+}
+
+std::string DynamicLibrary::GetFilename(const char* filename)
```

Comment:
Can (or should?) this use `GetFilename(const char*, int, int)` instead (or the other way around)?
Mostly to make sure they're not doing different things when no version number is passed.

Reply:
Good point. I've renamed the methods to be GetUnprefixedFilename() and GetVersionedFilename() to be clear about the behavior.